homeassistant/components/fritz/__init__.py

```diff
@@ -1 +1,63 @@
-"""The fritz component."""
+"""Support for AVM Fritz!Box functions."""
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .common import FritzBoxTools
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN, SUPPORTED_DOMAINS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up fritzboxtools from config entry."""
+    _LOGGER.debug("Setting up FRITZ!Box Tools component")
+    host = entry.data.get(CONF_HOST, DEFAULT_HOST)
+    port = entry.data.get(CONF_PORT, DEFAULT_PORT)
+    username = entry.data.get(CONF_USERNAME)
+    password = entry.data.get(CONF_PASSWORD)
+
+    fritz_tools = FritzBoxTools(
+        hass=hass,
+        host=host,
+        port=port,
+        username=username,
+        password=password,
+    )
+
+    try:
+        await fritz_tools.async_setup()
+    except FritzSecurityError as ex:
+        raise ConfigEntryAuthFailed from ex
+    except FritzConnectionException as ex:
+        raise ConfigEntryNotReady from ex
+
+    await fritz_tools.async_start()
```

Comment:
If this can throw, it should be in the `try`

Reply:
Once the connection is ok (this is done in the `try`), I don't see a way to throw for `async_start`

Reply:
What if the device goes offline between async_setup and async_start?